plugins/modules/ntnx_clusters.py

```diff
@@ -0,0 +1,522 @@
+        uplink_grouping=dict(type="str"),
+    )
+
+    network_spec = dict(
+        external_ip=dict(type="str"),
+        fully_qualified_domain_name=dict(type="str"),
+        external_data_services_ip=dict(type="str"),
+        external_subnet=dict(type="str"),
+        internal_subnet=dict(type="str"),
+        masquerading_ip=dict(type="str"),
+        masquerading_port=dict(type="str"),
+        domain_server=dict(type="dict", options=domain_server_spec),
+        nfs_subnet_whitelist=dict(type="list", elements="str"),
+        name_server_ip_list=dict(type="list", elements="str"),
+        ntp_server_ip_list=dict(type="list", elements="str"),
+        http_proxy_list=dict(type="list", elements="dict", options=http_proxy_list_spec),
+        smtp_server=dict(type="dict", options=smtp_server_spec),
+        http_proxy_whitelist=dict(type="list", elements="dict", options=http_proxy_whitelist_spec),
+        default_vswitch_config=dict(type="dict", options=default_vswitch_config_spec),
+    )
+    authorized_public_key_list_spec = dict(
+        name=dict(type="str"),
+        key=dict(type="str", no_log=False),
+    )
+
+    module_args = dict(
+        cluster=dict(
+            type="dict", options=entity_by_spec, mutually_exclusive=mutually_exclusive, required=True,
+        ),
+        authorized_public_key_list=dict(type="list", elements="dict", options=authorized_public_key_list_spec),
+        timezone=dict(type="str"),
+        supported_information_verbosity=dict(type="str"),
+        redundancy_factor=dict(type="int"),
+        network=dict(type="dict", options=network_spec),
+    )
+
+    return module_args
+
+
+def update_cluster_config(module, result):
+    cluster = Cluster(module)
+    old_spec = cluster.get_current_spec()
+    spec, error = cluster.get_spec(old_spec)
+    if error:
+        result["error"] = error
+        module.fail_json(msg="Failed generating cluster Spec", **result)
+
+    if module.check_mode:
+        result["response"] = spec
+        return
+    uuid = spec["metadata"]["uuid"]
+    resp = cluster.update(spec, uuid)
+    result["task_uuid"] = resp["status"]["execution_context"]["task_uuid"]
+
+    if module.params.get("wait"):
+        wait_for_task_completion(module, result)
+        resp = cluster.read(uuid)
+        result["response"] = resp
+
+    # compare the specs before and after.  currently only way to determine if something changed
+    if old_spec["spec"] != result["response"]["spec"]:
+        result["changed"] = True
+    else:
+        result["changed"] = False
+
+
+def wait_for_task_completion(module, result):
+    task = Task(module)
+    task_uuid = result["task_uuid"]
+    resp = task.wait_for_completion(task_uuid)
+    result["response"] = resp
+
+
+def run_module():
+    module = BaseModule(
+        argument_spec=get_module_spec(),
+        supports_check_mode=True,
+    )
+    remove_param_with_none_value(module.params)
+    result = {
```

Comment:
Add cluster_uuid as well. Its usefull when results of this module's task are used in other tasks in playbook